test/ambiguous.jl

```diff
@@ -162,6 +162,21 @@ end
 ambs = detect_ambiguities(Ambig48312)
 @test length(ambs) == 4
 
+module UnboundAmbig55868
+    module B
+        struct C end
+        export C
+        Base.@deprecate_binding D C
+    end
+    using .B
+end
+@test !Base.isbindingresolved(UnboundAmbig55868, :C)
+@test !Base.isbindingresolved(UnboundAmbig55868, :D)
+@test isempty(detect_unbound_args(UnboundAmbig55868))
+@test isempty(detect_ambiguities(UnboundAmbig55868))
+@test !Base.isbindingresolved(UnboundAmbig55868, :C)
+@test !Base.isbindingresolved(UnboundAmbig55868, :D)
```

Comment:
These tests seem to pass on nightly

```julia-repl
x@x:~$ julia +nightly --depwarn=error
A new `nightly` version is available. Install with `juliaup update`.
  o  | Version 1.12.0-DEV.1221 (2024-09-19)
 o o | Commit 58b239c5b8e (7 days old master)
julia> using Test

julia> module UnboundAmbig55868
           module B
               struct C end
               export C
               Base.@deprecate_binding D C
           end
           using .B
       end
Main.UnboundAmbig55868

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :C)
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :D)
Test Passed

julia> @test isempty(detect_unbound_args(UnboundAmbig55868))
Test Passed

julia> @test isempty(detect_ambiguities(UnboundAmbig55868))
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :C)
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :D)
Test Passed
```

Adding `export C, D` makes the issue reproduce, though:
```julia-repl
x@x:~$ julia +nightly --depwarn=error
A new `nightly` version is available. Install with `juliaup update`.
  o  | Version 1.12.0-DEV.1221 (2024-09-19)
 o o | Commit 58b239c5b8e (7 days old master)
julia> using Test

julia> module UnboundAmbig55868
           module B
               struct C end
               export C
               Base.@deprecate_binding D C
           end
           using .B
           export C, D
       end
Main.UnboundAmbig55868

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :C)
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :D)
Test Passed

julia> @test isempty(detect_unbound_args(UnboundAmbig55868))
WARNING: using deprecated binding B.D in UnboundAmbig55868.
, use C instead.
UnboundAmbig55868.D is deprecated, use UnboundAmbig55868.B.C instead.
Error During Test at REPL[5]:1
  Test threw exception
  Expression: isempty(detect_unbound_args(UnboundAmbig55868))
  use of deprecated variable: UnboundAmbig55868.D
  Stacktrace:
   [1] detect_unbound_args(mods::Any; recursive::Bool, allowed_undefineds::Nothing)
     @ Test ~/.julia/juliaup/julia-nightly/share/julia/stdlib/v1.12/Test/src/Test.jl:2170
   [2] detect_unbound_args(mods::Any)
     @ Test ~/.julia/juliaup/julia-nightly/share/julia/stdlib/v1.12/Test/src/Test.jl:2133
   [3] top-level scope
     @ REPL[5]:1
   [4] macro expansion
     @ ~/.julia/juliaup/julia-nightly/share/julia/stdlib/v1.12/Test/src/Test.jl:676 [inlined]
ERROR: There was an error during testing

julia> @test isempty(detect_ambiguities(UnboundAmbig55868))
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :C)
Test Failed at REPL[7]:1
  Expression: !(Base.isbindingresolved(UnboundAmbig55868, :C))

ERROR: There was an error during testing

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :D)
Test Failed at REPL[8]:1
  Expression: !(Base.isbindingresolved(UnboundAmbig55868, :D))

ERROR: There was an error during testing
```
And the PR fixes those failures
```julia-repl
x@x:~$ julia +pr55869 --depwarn=error
  o  | Version 1.12.0-DEV.1252 (2024-09-25)
 o o | devmotion:dw/test_deprecation/8694747bfed (fork: 2 commits, 0 days)
julia> using Test

julia> module UnboundAmbig55868
           module B
               struct C end
               export C
               Base.@deprecate_binding D C
           end
           using .B
           export C, D
       end
Main.UnboundAmbig55868

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :C)
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :D)
Test Passed

julia> @test isempty(detect_unbound_args(UnboundAmbig55868))
Test Passed

julia> @test isempty(detect_ambiguities(UnboundAmbig55868))
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :C)
Test Passed

julia> @test !Base.isbindingresolved(UnboundAmbig55868, :D)
Test Passed
```

Reply:
Oh no, good catch. Seems I tried too hard to reduce the MWE and forgot to rerun it on master. Will fix it.

Reply:
Done.